graphene_sqlalchemy/converter.py

```diff
@@ -26,29 +33,46 @@ def is_column_nullable(column):
     return bool(getattr(column, "nullable", True))
 
 
-def convert_sqlalchemy_relationship(relationship_prop, registry, connection_field_factory, resolver, **field_kwargs):
-    direction = relationship_prop.direction
-    model = relationship_prop.mapper.entity
-
+def convert_sqlalchemy_relationship(relationship_prop, obj_type, connection_field_factory, batching,
+                                    attr_name, orm_field_name, **field_kwargs):
+    """
+    :param sqlalchemy.RelationshipProperty relationship_prop:
+    :param Registry registry:
+    :type function|None connection_field_factory:
+    :type bool batching:
+    :param SQLAlchemyObjectType obj_type:
+    :param str orm_field_name:
+    :rtype: Dynamic
+    """
     def dynamic_type():
-        _type = registry.get_type_for_model(model)
+        direction = relationship_prop.direction
+        model = relationship_prop.mapper.entity
+        type_ = obj_type._meta.registry.get_type_for_model(model)
 
-        if not _type:
+        batching_ = batching if is_selectin_available else False
+        connection_field_factory_ = connection_field_factory
```

Comment:
Why do you need this allocation ?

Reply:
This is because I need to modify `connection_field_factory` in some cases (see [below](https://github.com/graphql-python/graphene-sqlalchemy/pull/260/files#r370926773)) and I can't reassign to `connection_field_factory` since it is defined outside of the scope `dynamic`. Otherwise, you get: `UnboundLocalError: local variable 'connection_field_factory' referenced before assignment`.